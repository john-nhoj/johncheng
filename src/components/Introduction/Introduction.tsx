import classNames from 'classnames';
import React from 'react';
import Image from 'next/image';

const Introduction = () => {
  return (
    <section
      className={classNames(
        'flex',
        'flex-col',
        'justify-center',
        'items-center',
        'hero-banner'
      )}
    >
      <Image
        src="/memoji.svg"
        alt="Memoji of John Cheng"
        width={190}
        height={190}
      />
      <div
        className={classNames(
          'flex',
          'justify-center',
          'mt-4',
          'text-4xl',
          'lg:text-5xl'
        )}
      >
        Hi, I'm John&nbsp;
        <span role="img" aria-label="Waving emoji">
          &#128075;
        </span>
      </div>
      <div
        className={classNames(
          'flex',
          'justify-center',
          'text-center',
          'mt-10',
          'text-2xl',
          'lg:text-3xl'
        )}
      >
        I am a Software Engineer and Badminton Enthusiast.
      </div>
      <style jsx={true}>{`
        .hero-banner {
          min-height: calc(100vh - 5rem);
        }
      `}</style>
    </section>
  );
};

export { Introduction };
