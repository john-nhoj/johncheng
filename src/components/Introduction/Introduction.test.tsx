import React from 'react';
import renderer from 'react-test-renderer';
import { Introduction } from 'components/Introduction/Introduction';

test('Introduction Section', () => {
  const header = renderer.create(<Introduction />);
  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();
});
