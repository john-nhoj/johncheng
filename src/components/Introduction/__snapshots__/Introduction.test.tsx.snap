// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction Section 1`] = `
<div
  className="flex flex-col justify-center w-full mt-16"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "margin": 0,
        "maxWidth": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "maxWidth": "100%",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkwIiBoZWlnaHQ9IjE5MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
        style={
          Object {
            "border": "none",
            "display": "block",
            "margin": 0,
            "maxWidth": "100%",
            "padding": 0,
          }
        }
      />
    </div>
    <noscript>
      <img
        alt="Memoji of John Cheng"
        decoding="async"
        src="/memoji.svg"
        srcSet="/_next/image?url=%2Fmemoji.svg&w=256&q=75 1x, /_next/image?url=%2Fmemoji.svg&w=384&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
    <img
      alt="Memoji of John Cheng"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="flex justify-center mt-4 text-4xl lg:text-5xl"
  >
    Hi, I'm JohnÂ 
    <span
      aria-label="Waving emoji"
      role="img"
    >
      ðŸ‘‹
    </span>
  </div>
  <div
    className="flex justify-center text-center mt-10 text-2xl lg:text-3xl"
  >
    I am a Software Engineer and Badminton Enthusiast.
  </div>
</div>
`;
